packages/commonwealth/server/controllers/server_groups_methods/create_group.ts

```diff
@@ -0,0 +1,52 @@
+import { ChainInstance } from 'server/models/chain';
+import { ServerChainsController } from '../server_chains_controller';
+import { AddressInstance } from '../../models/address';
+import { Requirement } from '../../util/requirementsModule/requirementsTypes';
+import { UserInstance } from '../../models/user';
+import validateRequirements from '../../util/requirementsModule/validateRequirements';
+import { AppError } from '../../../../common-common/src/errors';
+
+const Errors = {
+  InvalidRequirements: 'Invalid requirements',
+};
+
+export type CreateGroupOptions = {
+  user: UserInstance;
+  chain: ChainInstance;
+  address: AddressInstance;
+  metadata: any;
```

Comment:
Can we specify what a valid metadata object looks like?

Reply:
I've included the group metadata model in the migrations ticket. I'll update all ambiguous types to proper concrete types once the migration PR is merged.